### PR TITLE
Writes are accounted under FUSE_WRITE.

### PR DESCRIPTION
**Pull request by nitin-deamon**
Writes are accounted under FUSE_WRITE

### CODE REVIEW

File: turbonfs/inc/rpc_stats.h [hunk 1]
```diff
@@ -89,13 +89,8 @@ class rpc_stats_az
 
         /*
          * FUSE_RELEASE is sent as FUSE_FLUSH.
-         * Also, FUSE_FLUSH is accounted as FUSE_WRITE as both result in WRITE
-         * RPCs.
          */
         assert(_optype != FUSE_RELEASE);
-        if (_optype == FUSE_FLUSH) {
-            _optype = FUSE_WRITE;
-        }
```
Comment (linuxsmiths) [inline turbonfs/inc/rpc_stats.h:98]: Why did you remove this?
Note that aznfsc_ll_flush() still results in an RPC task of type FUSE_FLUSH to be created.

Comment (linuxsmiths) [inline turbonfs/inc/rpc_stats.h:98]: can't add the comment in on_rpc_issue(), hence adding it here.

Can you add the following in on_rpc_issue(), on_rpc_cancel(), on_rpc_complete(), 

// FUSE_FLUSH is never issued as an RPC task to the server. FUSE_WRITE is issued instead.
assert(optype != FUSE_FLUSH);

File: turbonfs/inc/rpc_stats.h [hunk 2]
```diff
@@ -231,6 +226,7 @@ class rpc_stats_az
                    optype == FUSE_READDIRPLUS ||
                    optype == FUSE_READ ||
                    optype == FUSE_WRITE ||
+                   optype == FUSE_FLUSH ||
```
Comment (linuxsmiths) [inline turbonfs/inc/rpc_stats.h:229]: add comment in L222

FUSE_FLUSH is never issued as an RPC to the server, so all FUSE_FLUSH tasks come here.

File: turbonfs/src/nfs_client.cpp [hunk 3]
```diff
@@ -457,11 +457,11 @@ void nfs_client::jukebox_runner()
                                 js->rpc_api->readdir_task.get_offset(),
                                 js->rpc_api->readdir_task.get_fuse_file());
                     break;
-                case FUSE_FLUSH:
+                case FUSE_WRITE:
                     AZLogWarn("[JUKEBOX REISSUE] flush(req={}, ino={})",
                               fmt::ptr(js->rpc_api->req),
                               js->rpc_api->flush_task.get_ino());
```
Comment (linuxsmiths) [inline turbonfs/src/nfs_client.cpp:463]: update this to

write(req={}, ino={}, bufv={}, off={}, len={}, fi={} [writepage: {}, flush: {}])

see aznfsc_ll_write_buf()

File: turbonfs/src/nfs_client.cpp [hunk 4]
```diff
@@ -1476,27 +1476,30 @@ void nfs_client::read(
...
-                           rpc_api->flush_task.get_ino());
+    assert(rpc_api->optype == FUSE_WRITE);
+
+    struct rpc_task *write_task =
+        get_rpc_task_helper()->alloc_rpc_task(FUSE_WRITE);
+    write_task->init_write(nullptr /* fuse_req */,
+                           rpc_api->write_task.get_ino(),
+                           nullptr /* bufv */,
+                           0 /* size */,
+                           0 /* offset */);
```
Comment (linuxsmiths) [inline turbonfs/src/nfs_client.cpp:1494]: why hard code these values? instead use whatever rpc_api->write_task has, so that the jukebox works fine regardless of whether it's called for user flush or write

File: turbonfs/src/rpc_stats.cpp [hunk 5]
```diff
@@ -237,6 +237,7 @@ do { \
     DUMP_OP(FUSE_READDIRPLUS);
     DUMP_OP(FUSE_READ);
     DUMP_OP(FUSE_WRITE);
+    DUMP_OP(FUSE_FLUSH);
```
Comment (linuxsmiths) [inline turbonfs/src/rpc_stats.cpp:240]: since we never issue RPC of this type to server, we will have 0 count for this

File: turbonfs/src/rpc_task.cpp [hunk 6]
```diff
@@ -942,9 +946,9 @@ bool rpc_task::add_bc(const bytes_chunk& bc)
...
 {
     // Must only be called for a flush task.
-    assert(get_op_type() == FUSE_FLUSH);
+    assert(get_op_type() == FUSE_WRITE);
 
-    const fuse_ino_t ino = rpc_api->flush_task.get_ino();
+    const fuse_ino_t ino = rpc_api->write_task.get_ino();
     struct nfs_inode *inode = get_client()->get_nfs_inode_from_ino(ino);
     struct bc_iovec *bciov = (struct bc_iovec *) rpc_api->pvt;
     assert(bciov->magic == BC_IOVEC_MAGIC);
```
Comment (linuxsmiths) [inline turbonfs/src/rpc_task.cpp:954]: add above this

// We should come here only for FUSE_WRITE tasks with bciov set.
assert(bciov != nullptr);

File: turbonfs/src/nfs_client.cpp [hunk 7]
```diff
@@ -1476,27 +1476,30 @@ void nfs_client::read(
...
  * rpc_api defines the RPC request that need to be retried.
  */
-void nfs_client::jukebox_flush(struct api_task_info *rpc_api)
+void nfs_client::jukebox_write(struct api_task_info *rpc_api)
 {
     /*
      * For write task pvt has write_iov_context, which has copy of byte_chunk vector.
      * To proceed it should be valid.
      */
     assert(rpc_api->pvt != nullptr);
```
Comment (linuxsmiths) [inline turbonfs/src/nfs_client.cpp:1485]: add after this

assert(((struct bc_iovec *) rpc_api->pvt)->magic == BC_IOVEC_MAGIC);

File: turbonfs/src/nfs_client.cpp (hunk 3 above)
Comment (linuxsmiths) [reply turbonfs/src/nfs_client.cpp:463]: Again, this confusion is due to the fact the we have two types of write tasks, one which correspond to user initiated values and has offset/buf/size valid and the rest which we issue to the backend, for which bc_iovec contains the data to be written

File: turbonfs/src/nfs_client.cpp (hunk 7 above)
Comment (nitin-deamon) [reply turbonfs/src/nfs_client.cpp:1485]: It's already done at L1496
